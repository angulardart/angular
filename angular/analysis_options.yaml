# This is a copy of ../analysis_options.yaml, with implicit-casts: false, and an "exclude: ...".

analyzer:
  strong-mode:
    implicit-casts: false
  exclude:
    - "tools/analyzer_plugin/bin/plugin.dart"
  errors:
    unused_element: error
    unused_import: error
    unused_local_variable: error
    dead_code: error

    # Allow importing .template.dart files without an analyzer error.
    uri_has_not_been_generated: ignore

    # The noise of deprecations isn't worth the effort.
    deprecated_member_use_from_same_package: ignore
linter:
  rules:
    - avoid_empty_else
    - avoid_init_to_null
    - avoid_null_checks_in_equality_operators
    - await_only_futures
    - camel_case_types
    - cancel_subscriptions
    - control_flow_in_finally
    - directives_ordering
    - empty_catches
    - empty_constructor_bodies
    - empty_statements
    - hash_and_equals
    - iterable_contains_unrelated_type
    - library_names
    - library_prefixes
    - list_remove_unrelated_type
    - only_throw_errors
    - overridden_fields
    - package_api_docs
    - package_names
    - package_prefixed_library_names
    - prefer_final_fields
    - prefer_is_not_empty
    - slash_for_doc_comments
    - test_types_in_equals
    - throw_in_finally
    - type_init_formals
    - unnecessary_brace_in_string_interps
    - unrelated_type_equality_checks
    - valid_regexps
