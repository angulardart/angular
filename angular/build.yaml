targets:
  $default:
    builders:
      angular:
        generate_for:
          exclude:
            - "lib/src/build.dart"
            - "lib/src/compiler/**"
            - "lib/src/source_gen/**"

builders:
  angular:
    import: "package:angular/src/build.dart"
    builder_factories:
      - templatePlaceholder
      - templateCompiler
      - stylesheetCompiler
    auto_apply: dependents
    applies_builders:
      - "angular|placeholder_cleanup"
      - "angular|component_source_cleanup"
    # See https://github.com/dart-lang/angular/issues/988.
    is_optional: true
    required_inputs:
      - ".css"
    build_extensions:
      .css:
        - ".css.dart"
        - ".css.shim.dart"
      .dart:
        - ".template.dart"
post_process_builders:
  placeholder_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "placeholderCleanup"
  component_source_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "componentSourceCleanup"
    defaults:
      generate_for: ["lib/**"]
      release_options:
        enabled: true
