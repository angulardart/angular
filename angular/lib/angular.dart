/// The primary library for the [AngularDart web framework][AngularDart].
///
/// Import this library as follows:
///
/// ```
/// import 'package:angular/angular.dart';
/// ```
///
/// For help using this library, see the AngularDart documentation:
///
/// * [AngularDart guide][]
/// * [AngularDart cheat sheet][cheatsheet]
///
/// [AngularDart]: https://webdev.dartlang.org/angular
/// [AngularDart guide]: https://webdev.dartlang.org/angular/guide
/// [cheatsheet]: https://webdev.dartlang.org/angular/cheatsheet

library angular;

export 'core.dart';
export 'src/common/common_directives.dart';
export 'src/common/directives.dart';
export 'src/common/pipes.dart';
export 'src/core/angular_entrypoint.dart' show AngularEntrypoint;
export 'src/core/application_tokens.dart' hide APP_ID_RANDOM_PROVIDER;
export 'src/platform/bootstrap.dart';
export 'src/platform/dom/dom_tokens.dart';
export 'src/platform/dom/events/event_manager.dart' show EventManagerPlugin;
