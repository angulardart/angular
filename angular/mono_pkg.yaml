dart:
- stable
- dev

stages:
- analyze:
  - dartanalyzer: --fatal-warnings .
