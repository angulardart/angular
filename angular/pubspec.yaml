name: angular
version: 5.3.0
author: Dart Team <web@dartlang.org>
description: Fast and productive web framework
homepage: https://webdev.dartlang.org/angular
documentation: https://webdev.dartlang.org/angular/api
environment:
  sdk: ">=2.2.1-dev.3.0 <3.0.0"

dependencies:
  # Runtime (or runtime *and* the compiler).
  collection: ^1.14.5
  intl: '>=0.13.0 <0.16.0'
  js: ^0.6.0
  logging: '>=0.9.0 <0.12.0'
  meta: ^1.1.4
  stack_trace: ^1.9.1

  # Compiler. Eventually we want to move these to angular_compiler.
  analyzer: ^0.36.2

  # It is extremely important these two dependencies stay pinned to current.
  angular_ast: 0.5.9
  angular_compiler: 0.4.3
  ##########################################################################

  build: '>=0.12.7 <2.0.0'
  build_config: '>=0.2.6 <0.5.0' # Runtime dependency with no import
  code_builder: '^3.0.1'
  csslib: '>=0.14.5 <0.16.0'
  path: ^1.6.1
  source_gen: ">=0.9.1 <0.10.0"
  source_span: ^1.4.0

# === vvv REMOVE WHEN PUBLISHING vvv ===
dependency_overrides:
  angular_ast:
      path: ../angular_ast
  angular_compiler:
      path: ../angular_compiler
# === ^^^ REMOVE WHEN PUBLISHING ^^^ ===
