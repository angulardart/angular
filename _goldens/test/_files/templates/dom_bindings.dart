import 'package:angular/angular.dart';

@Component(
  selector: 'has-host-attributes',
  template: '',
)
class HasHostAttributes {
  @HostBinding('attr.title')
  var title = 'Hello';
}

@Component(
  selector: 'has-template-attributes',
  template: r'''
    <div [attr.title]="title"></div>
  ''',
)
class HasTemplateAttributes {
  var title = 'Hello';
}

@Component(
  selector: 'has-static-attribute',
  template: r'''
    <div [attr.title]="'title'"></div>
  ''',
)
class HasStaticAttribute {}

@Component(
  selector: 'has-host-class',
  template: '',
)
class HasHostClass {
  @HostBinding('class.fancy')
  var fancy = true;
}

@Component(
  selector: 'has-template-class',
  template: r'''
    <div [class.fancy]="fancy"></div>
  ''',
)
class HasTemplateClass {
  var fancy = true;
}

@Component(
  selector: 'has-style-property',
  template: r'''
    <div
        [style.width]="width"
        [style.height.px]="height">
    </div>
  ''',
)
class HasStyleProperty {
  var width = 0;
  var height = 12;
}
