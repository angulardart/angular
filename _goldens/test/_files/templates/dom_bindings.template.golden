// **************************************************************************
// Generator: AngularDart Compiler
// **************************************************************************

import 'dom_bindings.dart';
export 'dom_bindings.dart';
import 'package:angular/src/di/reflector.dart' as _ngRef;
import 'package:angular/angular.template.dart' as _ref0;
import 'package:angular/src/core/linker/views/component_view.dart' as import0;
import 'dom_bindings.dart' as import1;
import 'package:angular/src/core/linker/style_encapsulation.dart' as import2;
import 'package:angular/src/core/linker/views/view.dart' as import3;
import 'package:angular/src/core/change_detection/change_detection.dart';
import 'dart:html' as import5;
import 'package:angular/src/runtime.dart' as import6;
import 'package:angular/src/runtime/dom_helpers.dart' as import7;
import 'package:angular/angular.dart';
import 'package:angular/src/core/linker/views/host_view.dart' as import9;

final List<dynamic> styles$HasHostAttributes = const [];

// View for component HasHostAttributes in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class ViewHasHostAttributes0 extends import0.ComponentView<import1.HasHostAttributes> {
  var _expr_0;
  static import2.ComponentStyles _componentStyles;
  ViewHasHostAttributes0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('has-host-attributes');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/templates/dom_bindings.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
  }

  void detectHostChanges(bool firstCheck) {
    final _ctx = this.ctx;
    final currVal_0 = _ctx.title;
    if (import6.checkBinding(this._expr_0, currVal_0, null, null)) {
      import7.updateAttribute(this.rootElement, 'title', currVal_0);
      this._expr_0 = currVal_0;
    }
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$HasHostAttributes, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.HasHostAttributes> _HasHostAttributesNgFactory = const ComponentFactory('has-host-attributes', viewFactory_HasHostAttributesHost0);
ComponentFactory<import1.HasHostAttributes> get HasHostAttributesNgFactory {
  return _HasHostAttributesNgFactory;
}

final List<dynamic> styles$HasHostAttributesHost = const [];

// Host for component HasHostAttributes in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class _ViewHasHostAttributesHost0 extends import9.HostView<import1.HasHostAttributes> {
  @override
  void build() {
    // Child component HasHostAttributes in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
    this.componentView = ViewHasHostAttributes0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.HasHostAttributes();
    this.initRootNode(_el_0);
  }

  @override
  void detectChangesInternal() {
    bool firstCheck = this.firstCheck;
    this.componentView.detectHostChanges(firstCheck);
    this.componentView.detectChanges();
  }
}

import9.HostView<import1.HasHostAttributes> viewFactory_HasHostAttributesHost0() {
  return _ViewHasHostAttributesHost0();
}

final List<dynamic> styles$HasTemplateAttributes = const [];

// View for component HasTemplateAttributes in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class ViewHasTemplateAttributes0 extends import0.ComponentView<import1.HasTemplateAttributes> {
  var _expr_0;
  import5.DivElement _el_0;
  static import2.ComponentStyles _componentStyles;
  ViewHasTemplateAttributes0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('has-template-attributes');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/templates/dom_bindings.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
    final doc = import5.document;
    this._el_0 = import7.appendDiv(doc, parentRenderNode);
  }

  @override
  void detectChangesInternal() {
    final _ctx = this.ctx;
    final currVal_0 = _ctx.title;
    if (import6.checkBinding(this._expr_0, currVal_0, 'title', '<FileLocation: 9 asset:_goldens/test/_files/templates/dom_bindings.dart:1:10>')) {
      import7.updateAttribute(this._el_0, 'title', currVal_0);
      this._expr_0 = currVal_0;
    }
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$HasTemplateAttributes, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.HasTemplateAttributes> _HasTemplateAttributesNgFactory = const ComponentFactory('has-template-attributes', viewFactory_HasTemplateAttributesHost0);
ComponentFactory<import1.HasTemplateAttributes> get HasTemplateAttributesNgFactory {
  return _HasTemplateAttributesNgFactory;
}

final List<dynamic> styles$HasTemplateAttributesHost = const [];

// Host for component HasTemplateAttributes in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class _ViewHasTemplateAttributesHost0 extends import9.HostView<import1.HasTemplateAttributes> {
  @override
  void build() {
    // Child component HasTemplateAttributes in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
    this.componentView = ViewHasTemplateAttributes0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.HasTemplateAttributes();
    this.initRootNode(_el_0);
  }
}

import9.HostView<import1.HasTemplateAttributes> viewFactory_HasTemplateAttributesHost0() {
  return _ViewHasTemplateAttributesHost0();
}

final List<dynamic> styles$HasStaticAttribute = const [];

// View for component HasStaticAttribute in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class ViewHasStaticAttribute0 extends import0.ComponentView<import1.HasStaticAttribute> {
  import5.DivElement _el_0;
  static import2.ComponentStyles _componentStyles;
  ViewHasStaticAttribute0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('has-static-attribute');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/templates/dom_bindings.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
    final doc = import5.document;
    this._el_0 = import7.appendDiv(doc, parentRenderNode);
  }

  @override
  void detectChangesInternal() {
    bool firstCheck = this.firstCheck;
    if (firstCheck) {
      import7.setAttribute(this._el_0, 'title', 'title');
    }
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$HasStaticAttribute, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.HasStaticAttribute> _HasStaticAttributeNgFactory = const ComponentFactory('has-static-attribute', viewFactory_HasStaticAttributeHost0);
ComponentFactory<import1.HasStaticAttribute> get HasStaticAttributeNgFactory {
  return _HasStaticAttributeNgFactory;
}

final List<dynamic> styles$HasStaticAttributeHost = const [];

// Host for component HasStaticAttribute in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class _ViewHasStaticAttributeHost0 extends import9.HostView<import1.HasStaticAttribute> {
  @override
  void build() {
    // Child component HasStaticAttribute in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
    this.componentView = ViewHasStaticAttribute0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.HasStaticAttribute();
    this.initRootNode(_el_0);
  }
}

import9.HostView<import1.HasStaticAttribute> viewFactory_HasStaticAttributeHost0() {
  return _ViewHasStaticAttributeHost0();
}

final List<dynamic> styles$HasHostClass = const [];

// View for component HasHostClass in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class ViewHasHostClass0 extends import0.ComponentView<import1.HasHostClass> {
  bool _expr_0;
  static import2.ComponentStyles _componentStyles;
  ViewHasHostClass0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('has-host-class');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/templates/dom_bindings.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
  }

  void detectHostChanges(bool firstCheck) {
    final _ctx = this.ctx;
    final currVal_0 = _ctx.fancy;
    if (import6.checkBinding(this._expr_0, currVal_0, null, null)) {
      import7.updateClassBindingNonHtml(this.rootElement, 'fancy', currVal_0);
      this._expr_0 = currVal_0;
    }
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$HasHostClass, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.HasHostClass> _HasHostClassNgFactory = const ComponentFactory('has-host-class', viewFactory_HasHostClassHost0);
ComponentFactory<import1.HasHostClass> get HasHostClassNgFactory {
  return _HasHostClassNgFactory;
}

final List<dynamic> styles$HasHostClassHost = const [];

// Host for component HasHostClass in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class _ViewHasHostClassHost0 extends import9.HostView<import1.HasHostClass> {
  @override
  void build() {
    // Child component HasHostClass in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
    this.componentView = ViewHasHostClass0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.HasHostClass();
    this.initRootNode(_el_0);
  }

  @override
  void detectChangesInternal() {
    bool firstCheck = this.firstCheck;
    this.componentView.detectHostChanges(firstCheck);
    this.componentView.detectChanges();
  }
}

import9.HostView<import1.HasHostClass> viewFactory_HasHostClassHost0() {
  return _ViewHasHostClassHost0();
}

final List<dynamic> styles$HasTemplateClass = const [];

// View for component HasTemplateClass in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class ViewHasTemplateClass0 extends import0.ComponentView<import1.HasTemplateClass> {
  bool _expr_0;
  import5.DivElement _el_0;
  static import2.ComponentStyles _componentStyles;
  ViewHasTemplateClass0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('has-template-class');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/templates/dom_bindings.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
    final doc = import5.document;
    this._el_0 = import7.appendDiv(doc, parentRenderNode);
  }

  @override
  void detectChangesInternal() {
    final _ctx = this.ctx;
    final currVal_0 = _ctx.fancy;
    if (import6.checkBinding(this._expr_0, currVal_0, 'fancy', '<FileLocation: 9 asset:_goldens/test/_files/templates/dom_bindings.dart:1:10>')) {
      import7.updateClassBinding(this._el_0, 'fancy', currVal_0);
      this._expr_0 = currVal_0;
    }
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$HasTemplateClass, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.HasTemplateClass> _HasTemplateClassNgFactory = const ComponentFactory('has-template-class', viewFactory_HasTemplateClassHost0);
ComponentFactory<import1.HasTemplateClass> get HasTemplateClassNgFactory {
  return _HasTemplateClassNgFactory;
}

final List<dynamic> styles$HasTemplateClassHost = const [];

// Host for component HasTemplateClass in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class _ViewHasTemplateClassHost0 extends import9.HostView<import1.HasTemplateClass> {
  @override
  void build() {
    // Child component HasTemplateClass in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
    this.componentView = ViewHasTemplateClass0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.HasTemplateClass();
    this.initRootNode(_el_0);
  }
}

import9.HostView<import1.HasTemplateClass> viewFactory_HasTemplateClassHost0() {
  return _ViewHasTemplateClassHost0();
}

final List<dynamic> styles$HasStyleProperty = const [];

// View for component HasStyleProperty in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class ViewHasStyleProperty0 extends import0.ComponentView<import1.HasStyleProperty> {
  var _expr_0;
  var _expr_1;
  import5.DivElement _el_0;
  static import2.ComponentStyles _componentStyles;
  ViewHasStyleProperty0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('has-style-property');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/templates/dom_bindings.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
    final doc = import5.document;
    this._el_0 = import7.appendDiv(doc, parentRenderNode);
  }

  @override
  void detectChangesInternal() {
    final _ctx = this.ctx;
    final currVal_0 = _ctx.width;
    if (import6.checkBinding(this._expr_0, currVal_0, 'width', '<FileLocation: 17 asset:_goldens/test/_files/templates/dom_bindings.dart:2:9>')) {
      this._el_0.style.setProperty('width', currVal_0?.toString());
      this._expr_0 = currVal_0;
    }
    final currVal_1 = _ctx.height;
    if (import6.checkBinding(this._expr_1, currVal_1, 'height', '<FileLocation: 47 asset:_goldens/test/_files/templates/dom_bindings.dart:3:9>')) {
      this._el_0.style.setProperty('height', ((currVal_1 == null) ? null : (currVal_1.toString() + 'px')));
      this._expr_1 = currVal_1;
    }
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$HasStyleProperty, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.HasStyleProperty> _HasStylePropertyNgFactory = const ComponentFactory('has-style-property', viewFactory_HasStylePropertyHost0);
ComponentFactory<import1.HasStyleProperty> get HasStylePropertyNgFactory {
  return _HasStylePropertyNgFactory;
}

final List<dynamic> styles$HasStylePropertyHost = const [];

// Host for component HasStyleProperty in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
class _ViewHasStylePropertyHost0 extends import9.HostView<import1.HasStyleProperty> {
  @override
  void build() {
    // Child component HasStyleProperty in asset:_goldens/test/_files/templates/dom_bindings.dart (changeDetection: Default)
    this.componentView = ViewHasStyleProperty0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.HasStyleProperty();
    this.initRootNode(_el_0);
  }
}

import9.HostView<import1.HasStyleProperty> viewFactory_HasStylePropertyHost0() {
  return _ViewHasStylePropertyHost0();
}

var _visited = false;
void initReflector() {
  if (_visited) {
    return;
  }
  _visited = true;

  _ngRef.registerComponent(HasHostAttributes, HasHostAttributesNgFactory);
  _ngRef.registerComponent(HasTemplateAttributes, HasTemplateAttributesNgFactory);
  _ngRef.registerComponent(HasStaticAttribute, HasStaticAttributeNgFactory);
  _ngRef.registerComponent(HasHostClass, HasHostClassNgFactory);
  _ngRef.registerComponent(HasTemplateClass, HasTemplateClassNgFactory);
  _ngRef.registerComponent(HasStyleProperty, HasStylePropertyNgFactory);
  _ref0.initReflector();
}
